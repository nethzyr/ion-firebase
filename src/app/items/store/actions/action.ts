import { Action } from '@ngrx/store';

export const LOAD_ITEMS = '[Items] Load items';

export class LoadItems implements Action {
  readonly type = LOAD_ITEMS;

  constructor(public payload: any) {}
}
