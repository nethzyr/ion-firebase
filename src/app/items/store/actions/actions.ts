import { Action } from '@ngrx/store';

export const LOAD_ITEMS = '[Items] Load Items';
export const LOAD_ITEMS_FAIL = '[Items] Load Items Fail';
export const LOAD_ITEMS_SUCCESS = '[Items] Load Items Success';

export class LoadItems implements Action {
  readonly type = LOAD_ITEMS;
}

export class LoadItemsFail implements Action {
  readonly type = LOAD_ITEMS_FAIL;
  constructor(public payload: string) {}
}

export class LoadItemsSuccess implements Action {
  readonly type = LOAD_ITEMS_SUCCESS;
  constructor(public payload: any) {}
}

export type ItemAction = LoadItems | LoadItemsFail | LoadItemsSuccess;
