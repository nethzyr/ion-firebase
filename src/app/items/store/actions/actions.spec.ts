import { LoadItems, LoadItemsFail, LoadItemsSuccess } from './actions';

describe('action', () => {
  test('LoadItems', () => {
    const action = new LoadItems();

    expect(action).toMatchSnapshot();
  });

  test('LoadItemsFail', () => {
    const action = new LoadItemsFail('randomId');

    expect(action).toMatchSnapshot();
  });

  test('LoadItemsSuccess', () => {
    const action = new LoadItemsSuccess('randomId');

    expect(action).toMatchSnapshot();
  });
});
