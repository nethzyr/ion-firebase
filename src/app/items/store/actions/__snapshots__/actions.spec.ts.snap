// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action LoadItems 1`] = `
LoadItems {
  "type": "[Items] Load Items",
}
`;

exports[`action LoadItemsFail 1`] = `
LoadItemsFail {
  "payload": "randomId",
  "type": "[Items] Load Items Fail",
}
`;

exports[`action LoadItemsSuccess 1`] = `
LoadItemsSuccess {
  "payload": "randomId",
  "type": "[Items] Load Items Success",
}
`;
